src/CairoLib.sol

```diff
@@ -161,19 +85,42 @@ library CairoLib {
         return staticcallCairo(contractAddress, functionSelector, data);
     }
 
-    /// @notice Performs a low-level call to a Cairo contract deployed on the Starknet appchain.
-    /// @dev Used with intent to read the state of the Cairo contract.
-    /// @param contractAddress The address of the Cairo contract.
-    /// @param functionName The name of the Cairo contract function to be called.
-    /// @return The return data from the Cairo contract function.
-    function staticcallCairo(uint256 contractAddress, string memory functionName)
+    function staticcallCairo(CairoCall memory call)
         internal
         view
         returns (bytes memory)
     {
-        uint256[] memory data = new uint256[](0);
-        uint256 functionSelector = uint256(keccak256(bytes(functionName))) % 2 ** 250;
-        return staticcallCairo(contractAddress, functionSelector, data);
+        return staticcallCairo(call.contractAddress, call.functionSelector, call.data);
+    }
+
+
+    /// @notice Performs a multicall to a Cairo contract deployed on Starknet.
+    /// @dev Used with intent to modify the state of the Cairo contract.
+    /// @param calls The array of CairoCall structs to be called.
+    function multicallCairo(CairoCall[] memory calls) internal {
+        uint256 n_calls = calls.length;
+        bytes memory callData = abi.encode(n_calls);
+        for (uint32 i = 0; i < n_calls; i++) {
+            bytes memory encodedCall = abi.encode(calls[i].contractAddress, calls[i].functionSelector, calls[i].data);
+            callData = bytes.concat(callData, encodedCall);
+        }
+        (bool success,) = MULTICALL_CAIRO_PRECOMPILE.call(callData);
+        require(success, "CairoLib: multicallCairo failed");
+    }
+
+    /// @notice Performs a multicall to a Cairo contract deployed on Starknet.
+    /// @dev Used with intent to read the state of the Cairo contract.
+    /// @dev **This can still mutate the underlying Cairo contract state.**
+    /// @param calls The array of CairoCall structs to be called.
```

Comment:
```suggestion
    /// @notice Performs a multicall to Cairo contracts deployed on Starknet.
    /// @dev Used with intent to read the state of the Cairo contract.
    /// @dev **This can still mutate the underlying Cairo contract state.**
    /// @param calls The array of CairoCall structs to be executed.
```